src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs

```diff
@@ -46,11 +47,21 @@ public RequestThrottlingMiddleware(RequestDelegate next, ILoggerFactory loggerFa
         /// <returns>A <see cref="Task"/> that completes when the request leaves.</returns>
         public async Task Invoke(HttpContext context)
         {
-            var waitInQueueTask = _requestQueue.EnterQueue();
+            var waitInQueueTask = _requestQueue.TryEnterQueueAsync();
 
             if (waitInQueueTask.IsCompletedSuccessfully)
             {
-                RequestThrottlingLog.RequestRunImmediately(_logger);
+                var requestPassedThrough = await waitInQueueTask;
```

Comment:
I know this can currently only be false in the IsCompletedSuccessfully case, but it is kinda weird to not check whether TryEnterQueueAsync's return value is true or false in the async changes. What if the implementation changes?